.github/workflows/regression-report.yml

```diff
@@ -122,24 +168,219 @@ jobs:
+          cat "$CONFIG_FILE_PATH"
+
+      - name: Upload Config and Project Files
+        uses: actions/upload-artifact@v3
+        with:
+          name: generated-config-and-project-file
+          path: |
+            ${{ steps.generate_config.outputs.config_link }}
+            ${{ steps.generate_config.outputs.projects_link }}
+
+  handle_configs_in_pr_description:
+    needs: [ parse_comment, checkout_and_cache ]
+    if: needs.parse_comment.outputs.mode == 'configs_in_pr_description'
+    runs-on: ubuntu-latest
+    outputs:
+      config_link: ${{ steps.parse_description.outputs.config_link }}
+      projects_link: ${{ steps.parse_description.outputs.projects_link }}
+    steps:
+      - name: Restore cached checkouts
+        uses: actions/cache@v4
+        with:
+          path: |
+            .
+            .ci-temp/checkstyle
+          key: ${{ runner.os }}-checkouts-${{ github.sha }}
+
+      - name: Getting PR description
+        env:
+          ISSUE_BODY: ${{ github.event.issue.body }}
+          PULL_REQUEST_URL: ${{ github.event.issue.pull_request.url }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          mkdir -p .ci-temp
+          # convert windows line endings to unix in event text
+          echo "$ISSUE_BODY" > .ci-temp/windows.txt
+          tr -d '\15\32' < .ci-temp/windows.txt > .ci-temp/text
+
+          curl --fail-with-body -X GET "${PULL_REQUEST_URL}" \
+              -H "Accept: application/vnd.github+json" \
+              -H "Authorization: token $GITHUB_TOKEN" \
+              -o .ci-temp/info.json
+
+          jq --raw-output .head.ref .ci-temp/info.json > .ci-temp/branch
+          jq --raw-output .head.sha .ci-temp/info.json > .ci-temp/commit_sha
+
+      - name: Parsing content of PR description
+        id: parse_description
+        run: |
+          ./.ci/diff-report.sh parse-pr-description-text
+
   make_report:
+    needs: [
+      parse_comment,
+      handle_config_bundle,
+      handle_generate_config,
+      handle_configs_in_pr_description
+    ]
+    if: |
+      always() &&
+      (needs.handle_config_bundle.result == 'success'
+        || needs.handle_generate_config.result == 'success'
+        || needs.handle_configs_in_pr_description.result == 'success')
     runs-on: ubuntu-latest
-    needs: parse_comment
-    if: needs.parse_comment.outputs.config_link != ''
-      || needs.parse_comment.outputs.projects_link != ''
     outputs:
       message: ${{ steps.out.outputs.message }}
       download_files_failed: ${{ steps.download_files.outputs.failed }}
     steps:
-
       - name: Download checkstyle
         uses: actions/checkout@v4
 
-      - name: Download files
+      - name: Download generated config artifacts
+        if: needs.handle_generate_config.result == 'success'
+        uses: actions/download-artifact@v3
+        with:
+          name: generated-config-and-project-file
+          path: .generated-configs
```

Comment:
just get files by cache.
and you will not need to do 
```
 - name: Download checkstyle
   uses: actions/checkout@v4
```

Reply:
@romani not resolved, looking into it. mentioning as it says outdated.

Reply:
@romani, i had mentioned this before. I will add the notes. in make_report, we need to checkout. 
```
The cached version of the repository doesn't include the GitHub Actions-specific Git configurations and credentials that are typically set up by the actions/checkout action. When you used the actions/checkout@v4 action, it performed a fresh checkout. This action automatically sets up the Git environment correctly, including authentication, which is why it worked.

However, 
When you restore the cache in the make_report job, we are bringing back the entire repository state, including the .git directory. However, the Git configuration might not be fully preserved in the cache. 

There are alternatives like add configure git operations in the make_report. For the sake of simplicity can we do just one checkout of the checkstyle in the whole report at make_report. rest use the cached version.
```

Reply:
If you see other jobs, we don't checkout. But, for this make_report, i had to checkout leaving no alternative.

Reply:
And, if i try to use the cache, the step for `set_upstream `fails. I can add the cache and mention the runner highlight the filaure to add substance to the above comment.

Reply:
>When you used the actions/checkout@v4 action, it performed a fresh checkout. This action automatically sets up the Git environment correctly, including authentication, which is why it worked.

Please add this to comment above checkout , it is not very obvious nuance.

Reply:
https://github.com/checkstyle/checkstyle/issues/15554 and updated the workflow with # until ...

Reply:
I made a comment, please consider

Reply:
done. removed all checkouts including checkstyle/contribution. now, just relying on cache.